### PR TITLE
Update mongo versions to latest releases

### PR DESCRIPTION
**Pull request by dmoody256**

### CODE REVIEW

File: Formula/mongodb-csfle.rb [hunk 1]
```diff
@@ -5,13 +5,11 @@ class MongodbCsfle < Formula
   license "MongoDB Customer Agreement"
 
   if Hardware::CPU.arm?
-    # TODO(MONGOSH-1193): Bump to production version after 6.0 is released
-    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-arm64-enterprise-6.0.0-rc4.tgz"
-    sha256 "5e5bc88bcadf256c7dcea08a7aa55a552630de6e6919fcf3215b5e8171231c03"
+    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-arm64-enterprise-6.0.1.tgz"
```
Comment (addaleax) [inline Formula/mongodb-csfle.rb:8]:
```suggestion
    url "https://downloads.mongodb.com/osx/mongo_crypt_shared_v1-macos-arm64-enterprise-6.0.1.tgz"
```

File: Formula/mongodb-csfle.rb [hunk 2]
```diff
@@ -5,13 +5,11 @@ class MongodbCsfle < Formula
...
+    sha256 "be79ec1cbb12420031f1aa7bb8594e5c88a81b7ab326e4f6a4ba7c64cf944c6d"
   else
-    # TODO(MONGOSH-1193): Bump to production version after 6.0 is released
-    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-x86_64-enterprise-6.0.0-rc4.tgz"
-    sha256 "4f1043d8b07ace6032a52eb41b193b3380061a7564ce21ffae24b1ccc0e5c558"
+    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-x86_64-enterprise-6.0.1.tgz"
```
Comment (addaleax) [inline Formula/mongodb-csfle.rb:11]:
```suggestion
    url "https://downloads.mongodb.com/osx/mongo_crypt_shared_v1-macos-x86_64-enterprise-6.0.1.tgz"
```

**Review by addaleax — COMMENTED** [review]
I think we forgot to update this repo as part of the crypt shared library renaming :)

**Review by IamXander — APPROVED** [review]

File: Formula/mongodb-csfle.rb (hunk 1 above)
Comment (dmoody256) [reply Formula/mongodb-csfle.rb]: Thanks for checking @addaleax! I updated in the latest rebase. You can close MONGOSH-1193 after this.

**Review by addaleax — APPROVED** [review]

**Review by deriamis — APPROVED** [review]

**Pull request merged**